pkg/server/resource_lister.go

```diff
@@ -9,39 +9,69 @@ import (
 	"github.com/openshift/console/pkg/serverutils"
 )
 
-// ResourceLister determines the list of resources of a particular kind
-type ResourceLister struct {
-	BearerToken string
-	RequestURL  *url.URL
-	Client      *http.Client
+// ResourceLister handles resource requests
+type ResourceLister interface {
+	HandleResources(w http.ResponseWriter, r *http.Request)
 }
 
-func (l *ResourceLister) handleResources(w http.ResponseWriter, r *http.Request) {
+// FilterFunction shall filter response before propagating
+type FilterFunction func(io.Writer, io.Reader) (int64, error)
```

Comment:
the original signature was correct

Reply:
Since we assign io.Copy to responseFilter, FilterFunction need to have same signature as of io.Copy.